### DISCUSSION TITLE
Generic TuyaMCU device to support Lellki power strip

### DISCUSSION THREAD

**Discussion by Rainerlan** [arendst/Tasmota#11480] [Q&A] [answered]
I am trying to get a Lellki power strip to work (Tuya).
After opening it, I know all the signals between TYWE3S, MCU, switches relay, etc. (See mapping below)
It seems to be impossible to configure Tasmota accordingly
Using Tasmota TuyaMCU (54) as basis, 
- Tasmota uses pre-defined commands for realais, buttons. etc.
- Tasmota seems to use GPIO0 some dedicated/pre-defined function 

As a result, mapping TuyaTX to GPIO0 does not work - I can't see any Tuya serial command on GPIO0 (Not even the heartbeat)

What I would need is a config, where the tuya commands are not predefined (sent on TuyaMCU_Tx), but where I can try-out the TuyaMCU commands.
I can't use any of the pre-configured Tuya-commands for my device (The Lellki device doesn't work this way).
Instead, with the Lellki power strip, 
- except for the 5 control buttons and 1 status LED, everything is connected to the TYWE3S directly (and controlled by it - eg. the relays)
- The MCU ist just needed to "code" the 5 switch keys, control the blue status LED and send a trigger for any switch-change to the Wifi module

**Q:** Is there any way to configure Tasmota in a way, that I can freely assign the TuyaRx/Tx w/o any dependencies on the "Tasmota-GPIOs"?

Pinout/connectinos on PCB:
TuyaTx on GPIO0
1 button-switch-trigger signal from MCU: connected to GPIO1
HWL8012 CF: Connected to GPIO4
Voltage-sense: Connexed to GPIO5
HWL8012 SEL: Connected to GPIO12
Relays1-4: Connected to GPIO3, 13, 14, 15
TuyaRx on GPIO16
5 Switches + 1 LED: Connected to MCU - status to be sent to TYWES3

**Comment by ascillato2** [comment, marked as answer]
Please, address this to the [Tasmota Support Chat](https://discord.gg/Ks2Kzd4). The chat is a better and more dynamic channel for helping you.Thanks.

---------------------------

### Support Information 
See [Docs](https://tasmota.github.io/docs/) for more information.
See [Chat](https://discord.gg/Ks2Kzd4) for more user experience.
See [Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
